go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -611,15 +655,23 @@ func (c *cnxn) Close() error {
 //
 // A partition can be retrieved by using ExecutePartitions on a statement.
 func (c *cnxn) ReadPartition(ctx context.Context, serializedPartition []byte) (rdr array.RecordReader, err error) {
-	var endpoint flight.FlightEndpoint
-	if err := proto.Unmarshal(serializedPartition, &endpoint); err != nil {
+	var info flight.FlightInfo
+	if err := proto.Unmarshal(serializedPartition, &info); err != nil {
 		return nil, adbc.Error{
 			Msg:  err.Error(),
 			Code: adbc.StatusInvalidArgument,
 		}
 	}
 
-	rdr, err = doGet(ctx, c.cl, &endpoint, c.clientCache)
+	// The driver only ever returns one endpoint.
```

Comment:
this isn't a guarantee though? The number of endpoints would be determined by the server it's communicating with, wouldn't it?

Reply:
A serialized partition from this driver always has exactly 1 endpoint. But I want to make the serialized partition contain the full FlightInfo otherwise since the client might want to extract the other Flight-specific metadata there.